### ISSUE TITLE
Problems with share dialog with Facebook SDK v12/v13

### ISSUE THREAD

**Issue by paulocoutinhox** [facebook/facebook-android-sdk#1081]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [Github issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Java version

8

### Android version

Xiaomi - Android 12

### Android SDK version

13.2

### Installation platform & version

Gradle

### Package

Share

### Goals

I want show a share dialog like i made in iOS (that is working there)

### Expected results

Show the share dialog with my photo

### Actual results

It shows only an error message on log:

```
[ShareStoryForProductViewModel : onSharePressed] Error when share story: Parameter specified as non-null is null: method X.0zc.A1E, parameter param at index = 0
```

### Code samples & details

```java
val imageUri = Uri.parse(imagePath)
val photo = SharePhoto.Builder().setImageUrl(imageUri).build()
val backgroundColorList = item.colors.toList()

val content = ShareStoryContent.Builder()
    .setStickerAsset(photo)
    .setBackgroundColorList(backgroundColorList)
    .build()

val dialog = ShareDialog(context as BaseActivity)
dialog.registerCallback(
    callbackManager,
    object : FacebookCallback<Sharer.Result> {
        override fun onSuccess(result: Sharer.Result) {
            Logger.d("[ShareStoryForProductViewModel : onSharePressed] Share story posted with success")
        }

        override fun onCancel() {
            Logger.d("[ShareStoryForProductViewModel : onSharePressed] Share story post canceled")
        }

        override fun onError(error: FacebookException) {
            Logger.e("[ShareStoryForProductViewModel : onSharePressed] Error when share story: $error")
        }
    })

if (ShareDialog.canShow(SharePhotoContent::class.java)) {
    dialog.show(content)
}
```

**Comment by paulocoutinhox** [comment]
I tried with intent too: https://developers.facebook.com/docs/sharing/sharing-to-stories/android-developers

But Facebook is crashing:

```
val imageUri = Uri.parse("file://$imagePath")
val appId = Kite.string[R.string.facebook_app_id]

val intent = Intent("com.facebook.stories.ADD_TO_STORY")
intent.type = "image/png"
intent.putExtra("com.facebook.platform.extra.APPLICATION_ID", appId)
intent.putExtra("interactive_asset_uri", imageUri)
intent.putExtra("top_background_color", item.colors[0])
intent.putExtra("bottom_background_color", item.colors[1])

val activity = (context as BaseActivity)
activity.grantUriPermission(
    "com.facebook.katana",
    imageUri,
    Intent.FLAG_GRANT_READ_URI_PERMISSION
)

if (activity.packageManager.resolveActivity(intent, 0) != null) {
    activity.startActivityForResult(intent, 0)
}
```

![fb1](https://user-images.githubusercontent.com/395096/173880243-82420a9c-3b27-4607-a377-77896168ff3a.png)
![fb2](https://user-images.githubusercontent.com/395096/173880256-659f87fa-34be-470d-a384-b2a71085eedc.png)

**Comment by paulocoutinhox** [comment]
I receive other type of error:

```
Unable to show the provided content via the web or the installed version of the Facebook app. Some dialogs are only supported starting API 14.
```

**Comment by angiep125** [comment]
I would recommend sticking with the Share to Stories Intent API. 
What app version is your Facebook App? Does it still crash on latest?

**Comment by paulocoutinhox** [comment]
Hi,

Im using the latest app from store:

![Screenshot_20220617_133618](https://user-images.githubusercontent.com/395096/174340402-778adf4f-f9ac-4111-bc08-9632ca27f3af.png)

**Comment by paulocoutinhox** [comment]
I made a video:

https://user-images.githubusercontent.com/395096/174341362-54434d42-d3bc-4f70-96be-52eb574f67e4.mp4

My intent:

```
shareIntent = Intent("com.facebook.stories.ADD_TO_STORY").apply {
    type = "image/png"
    flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
    putExtra("source_application", Application.instance.packageName)
    putExtra("com.facebook.platform.extra.APPLICATION_ID", appId)
    putExtra("interactive_asset_uri", stickerAssetUri)
    putExtra("top_background_color", item.colors[0])
    putExtra("bottom_background_color", item.colors[1])
}
```

<img width="875" alt="image" src="https://user-images.githubusercontent.com/395096/174341676-c202d63c-5046-49c0-b95b-2c67446d88aa.png">

**Comment by paulocoutinhox** [comment]
I made a new project, showing only this featured, with code and video, simple to debug now:

https://github.com/paulocoutinhox/android-share-story/issues/1


https://user-images.githubusercontent.com/395096/174397492-8fcd797c-dfa4-495b-84dd-167bedf70312.mp4



Same problem happen.

**Comment by angiep125** [comment]
Hmm, I wonder if your file Uri is inaccessible to FB app. have you tried using
FileProvider.getUriForFile(this, getPackageName(), imageFile)
https://ali-dev.medium.com/open-a-file-in-another-app-with-android-fileprovider-for-android-7-42c9abb198c1

**Comment by paulocoutinhox** [comment]
Hi,

It works. I updated the sample project.

I have other questions to make image background and vide background works.

To image and video background i will need save it locally and get the uri for this too? Example:

```
shareIntent = Intent("com.facebook.stories.ADD_TO_STORY").apply {
    setDataAndType(backgroundAssetUri, "image/png") // it need be shared by provider in the same way?
    flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
    putExtra("source_application", Application.instance.packageName)
    putExtra("com.facebook.platform.extra.APPLICATION_ID", appId)
    putExtra("interactive_asset_uri", stickerAssetUri)
}
``` 

Thanks.

**Comment by angiep125** [comment]
Awesome! Yeap, it will also need a local path file and will need to use FileProvider to generate the Uri to pass in the intent. 
Please comment again here if you have issues with the image/video background asset to avoid re-triaging new issue :) 

I'll work on getting the sample docs updated to switch to using FileProvider instead of Uri.parse to unblock future developers.

**Comment by paulocoutinhox** [comment]
And on Instagram the image is stretched and the video is squared too. There is any way to solve it?

![Screenshot_20220621_133746](https://user-images.githubusercontent.com/395096/174852593-8afb20b3-417a-4326-9c4c-718802f20f0c.png)
![Screenshot_20220621_133729](https://user-images.githubusercontent.com/395096/174852601-75f172b5-5af1-468e-9445-999e48f1ca9c.png)

**Comment by angiep125** [comment]
Please open new case for instagram support. 
Can you verify what the dimensions & size are for the media shared to facebook? What happens if you share the Image to stories directly from gallery (rather than from app)? Is it still stretched horizontally?
Docs specify the following for the media:
"Uri to an image (JPG, PNG) or video (H.264, H.265, WebM). Recommended image dimensions: device fullscreen or smaller. Videos can be 1080p and up to 20 seconds in duration. The Uri needs to be a content Uri to a local file on device."

**Comment by paulocoutinhox** [comment]
There is any way to reduce the image size on IG/FB? Because it starts with a big size. There is any param on intent to reduce it  to 100 of width (example)?

**Comment by paulocoutinhox** [comment]
Problem solved. Only need convert to correct format in Handbrake software. Thanks.

**Comment by paulocoutinhox** [comment]
The only problem remaining is the size of image on the apps. The FB & IG don't show the image as it size, but bigger.

The image is small, but the apps show it big a lot. There is any way to make the apps show the image as it size?

Example:


![Screenshot_20220623_181032](https://user-images.githubusercontent.com/395096/175400629-42ab2fad-9712-4023-8b24-5ac0b4950132.png)
![Screenshot_20220623_181054](https://user-images.githubusercontent.com/395096/175400568-38df109e-c65d-4cb6-83a2-8e513e8a46e3.png)

**Comment by paulocoutinhox** [comment]
Ok. To me this issue can be closed. Everything is working. I have updated the sample with all code.

**Comment by Raymond9Chevreux** [comment]
I always respect people who express themselves as personalities. Anyway, I've been thinking about popularity among parents for so long. Almost every one of them also has a good number of likes on their posts and photos, but I didn't need to use it before. Nevertheless, once I decided to buy shares on this source https://top4smm.com/facebook-shares where you can do the same thing. Good luck.
